src/NuGet.Core/NuGet.LibraryModel/ToolDependency.cs

```diff
@@ -1,14 +1,47 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using NuGet.Frameworks;
+using NuGet.Shared;
 
 namespace NuGet.LibraryModel
 {
-    public class ToolDependency
+    public class ToolDependency : IEquatable<ToolDependency>
```

Comment:
Decided to do this as a separate item:
https://github.com/NuGet/Home/issues/3986